eureka-client/src/main/java/com/netflix/discovery/provider/DiscoveryJerseyProvider.java

```diff
@@ -44,92 +43,81 @@
  * server.
- * <p/>
- * <p>
- * This implementation allows users to plugin their own
- * serialization/deserialization mechanism by reading the annotation provided by
- * specifying the {@link Serializer} and dispatching it to that implementation.
- * </p>
  *
  * @author Karthik Ranganathan
  */
 @Provider
-@Produces("*/*")
+@Produces({"application/json", "application/xml"})
 @Consumes("*/*")
-public class DiscoveryJerseyProvider implements MessageBodyWriter, MessageBodyReader {
+public class DiscoveryJerseyProvider implements MessageBodyWriter<Object>, MessageBodyReader<Object> {
     private static final Logger LOGGER = LoggerFactory.getLogger(DiscoveryJerseyProvider.class);
 
-    // Cache the serializers so that they don't have to be instantiated every time
-    private static ConcurrentHashMap<Class, ISerializer> serializers = new ConcurrentHashMap<Class, ISerializer>();
+    private final EncoderWrapper jsonEncoder;
+    private final DecoderWrapper jsonDecoder;
 
-    private final EncoderWrapper encoder;
-    private final DecoderWrapper decoder;
+    // XML support is maintained for legacy/custom clients. These codecs are used only on the server side only
+    private final EncoderWrapper xmlEncoder;
+    private final DecoderWrapper xmlDecoder;
 
     public DiscoveryJerseyProvider() {
         this(null, null);
     }
 
-    public DiscoveryJerseyProvider(EncoderWrapper encoder, DecoderWrapper decoder) {
-        this.encoder = encoder == null ? CodecWrappers.getEncoder(LegacyJacksonJson.class) : encoder;
-        this.decoder = decoder == null ? CodecWrappers.getDecoder(LegacyJacksonJson.class) : decoder;
+    public DiscoveryJerseyProvider(EncoderWrapper jsonEncoder, DecoderWrapper jsonDecoder) {
+        this.jsonEncoder = jsonEncoder == null ? CodecWrappers.getEncoder(LegacyJacksonJson.class) : jsonEncoder;
+        this.jsonDecoder = jsonDecoder == null ? CodecWrappers.getDecoder(LegacyJacksonJson.class) : jsonDecoder;
+        LOGGER.info("Using JSON encoding codec {}", this.jsonEncoder.codecName());
+        LOGGER.info("Using JSON decoding codec {}", this.jsonDecoder.codecName());
 
-        if (encoder instanceof CodecWrappers.JacksonJsonMini) {
-            throw new UnsupportedOperationException("Encoder: " + encoder.codecName() + "is not supported for the client");
+        if (jsonEncoder instanceof CodecWrappers.JacksonJsonMini) {
+            throw new UnsupportedOperationException("Encoder: " + jsonEncoder.codecName() + "is not supported for the client");
         }
 
-        LOGGER.info("Using encoding codec {}", this.encoder.codecName());
-        LOGGER.info("Using decoding codec {}", this.decoder.codecName());
-    }
-
-    public EncoderWrapper getEncoder() {
-        return encoder;
-    }
+        // If jackson-dataformat-xml available, initialize XML codecs
+        if (isJacksonXmlOnClasspath()) {
+            this.xmlEncoder = CodecWrappers.getEncoder("JacksonXml");

```

Comment:
This might cause ClassNotFoundException if jackson-dataformat-xml is not on classpath.